Bump version to 0.0.1-alpha.34 (#1868)
astral-sh/ty: CHANGELOG.md

# Changelog

## 0.0.1-alpha.34

Released on 2025-12-12.

### Bug fixes

- Fix outdated version in publish diagnostics after `didChange` ([#21943](https://github.com/astral-sh/ruff/pull/21943))
- Fix workspace symbols to return members too ([#21926](https://github.com/astral-sh/ruff/pull/21926))

> - Respect typevar bounds when matching against a union ([#21893](https://github.com/astral-sh/ruff/pull/21893))

Reviewer: ```suggestion
- Improve solving of a type variable with an upper bound when that type variable appears as one element in a union type ([#21893](https://github.com/astral-sh/ruff/pull/21893))
```

- Respect `kw_only` from parent class ([#21820](https://github.com/astral-sh/ruff/pull/21820))
- Avoid inferring types for invalid binary expressions in string annotations ([#21911](https://github.com/astral-sh/ruff/pull/21911))
- Fix disjointness checks on `@final` class instances ([#21769](https://github.com/astral-sh/ruff/pull/21769))
- Fix disjointness checks with type-of `@final` classes ([#21770](https://github.com/astral-sh/ruff/pull/21770))
- Avoid fixpoint unioning of types containing current-cycle Divergent ([#21910](https://github.com/astral-sh/ruff/pull/21910))
- Avoid unions of generic aliases of the same class in fixpoint ([#21909](https://github.com/astral-sh/ruff/pull/21909))

### LSP server

- Adjust scope completions to use all reachable symbols ([#21872](https://github.com/astral-sh/ruff/pull/21872))
- Classify `cls` as class parameter for semantic highlighting ([#21944](https://github.com/astral-sh/ruff/pull/21944))

> - Don't show hover for expressions with no inferred type ([#21924](https://github.com/astral-sh/ruff/pull/21924))

Reviewer: ```suggestion
- Don't show on-hover tooltips for expressions with no inferred type ([#21924](https://github.com/astral-sh/ruff/pull/21924))
```

- Ignore `__all__` for document and workspace symbol requests ([#21928](https://github.com/astral-sh/ruff/pull/21928))
- Recognize `__all__ += submodule.__all__` in auto-import ([#21918](https://github.com/astral-sh/ruff/pull/21918))
- Stabilize rename ([#21940](https://github.com/astral-sh/ruff/pull/21940))

### Other changes

> - Support checking files without extension ([#21867](https://github.com/astral-sh/ruff/pull/21867))

Reviewer: ```suggestion
- Support checking files without extensions ([#21867](https://github.com/astral-sh/ruff/pull/21867))
```

- Defer all parameter and return type annotations ([#21906](https://github.com/astral-sh/ruff/pull/21906))
- Improve resolution of absolute imports in tests ([#21817](https://github.com/astral-sh/ruff/pull/21817))
- Infer type of implicit `cls` parameter in method bodies ([#21685](https://github.com/astral-sh/ruff/pull/21685))
- Support implicit type of `cls` in signatures ([#21771](https://github.com/astral-sh/ruff/pull/21771))
- Uniformly use "not supported" in diagnostics ([#21916](https://github.com/astral-sh/ruff/pull/21916))
- Add `SyntheticTypedDictType` and implement `normalized` and `is_equivalent_to` ([#21784](https://github.com/astral-sh/ruff/pull/21784))
- Default-specialize class-literal types in assignment to generic-alias types ([#21883](https://github.com/astral-sh/ruff/pull/21883))

> - Improve bad specialization results & error messages ([#21840](https://github.com/astral-sh/ruff/pull/21840))

Author: What have you got against the `&`!?!?! Haha
